src/System.Net.Security/tests/Scripts/setup-kdc.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+OS=`cat /etc/os-release | grep "^ID=" | sed 's/ID=//g' | sed 's/["]//g' | awk '{print $1}'`
+echo "Operating System: ${OS}"
+
+realm="TEST.COREFX.NET"
+
+principal1="HOST/host.test.corefx.net"
+principal2="HTTP"
+krb_user="krb_user"
+password="password"
+
+kadmin="kadmin.local"
+krb5kdc="krb5kdc"
+kdb5_util="kdb5_util"
+add_principal_cmd="add_principal -pw ${password}"
+
+krb_conf="krb5.conf"
+krb_conf_location="/etc/"
+keytabfile="/etc/krb5.keytab"
+
+# Common function across linux distro's to configure KDC post installation
+configure_kdc()
+{
+	echo "Stopping KDC.."
+	if pgrep krb5kdc 2> /dev/null; then sudo killall krb5kdc ; fi
+	
+	# Remove database files if exist
+	# There are multiple kdc db files with name pattern principal*. Doing "sudo rm -f" doesn't work,
+	# because the file name expansion happens in context of normal user, (who does not have pemission on the path)
+	# and then the command is passed to sudo. Hence it tries to delete file with name "principal*", which does not exist.
+	# To overcome this and make file name expansion under sudoer/superuser, rm is run under a separate bash 
+	sudo bash -c "rm -f ${database_files}"
+
+	# Create/copy krb5.conf and kdc.conf
+	echo "Copying krb5.conf and kdc.conf.."
+	sudo /bin/cp ${krb_conf} ${krb_conf_location}
+	sudo /bin/cp ${kdc_conf} ${kdc_conf_location}
+	
+	echo "Creating KDC database for realm ${realm}.."
+	sudo ${kdb5_util} create -r ${realm} -P ${password} -s
+	
+	echo "Adding principal ${principal1}.."
+	sudo ${kadmin} -q "${add_principal_cmd} ${principal1}@${realm}"
+	
+	echo "Adding principal ${principal2}.."
+	sudo ${kadmin} -q "${add_principal_cmd} ${principal2}@${realm}"
+	
+	echo "Adding user ${krb_user}.."
+	sudo ${kadmin} -q "${add_principal_cmd} ${krb_user}@${realm}"
+	
+	echo "Exporting keytab for ${principal1}"
+	sudo ${kadmin} -q "ktadd -norandkey ${principal1}@${realm}"
+	
+	echo "Exporting keytab for ${principal2}"
+	sudo ${kadmin} -q "ktadd -norandkey ${principal2}@${realm}"
+	
+	echo "Exporting keytab for ${krb_user}"
+	sudo ${kadmin} -q "ktadd -norandkey ${krb_user}@${realm}"
+}
+
+case ${OS} in
+	"ubuntu" | "debian")
+		kdc_conf="kdc.conf.ubuntu"
+		kdc_conf_location="/etc/krb5kdc/kdc.conf"
+		database_files="/var/lib/krb5kdc/principal*"
+		
+		dpkg -s krb5-kdc >/dev/null 2>&1
+		if [ $? -ne 0 ]
+		then 
+			echo "Installing krb5-kdc.."
+			sudo DEBIAN_FRONTEND=noninteractive apt-get -y install krb5-kdc krb5-admin-server
+			if [ $? -ne 0 ]
+			then
+				echo "Error occurred during installation, aborting"
+				exit 1
+			fi
+		else
+			echo "krb5-kdc already installed.."
+			exit 0
+		fi
+		
+		configure_kdc
```

Comment:
@stephentoub we are not actually overwriting user's setup. If the kdc is already installed, we are assuming that the machine is already setup with required users for the testing (through earlier runs of the script). If this assumption is indeed correct, then we might not require the force switch. 
Let me know if this assumption is correct.